submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/handlers/FilesHandler.kt

```diff
@@ -25,32 +32,52 @@ class FilesHandler(private val folderResolver: FolderResolver, private val seria
         val fileSource = if (files.isEmpty()) PathFilesSource(getUserFolder(submission)) else ListFilesSource(files)
 
         validateFiles(submission, fileSource)
-        copyFiles(submission, fileSource)
-        generateOutputFiles(submission)
-    }
+        copyFiles(submission.allFiles(), submission, fileSource)
+
+        validateLibraryFiles(submission, fileSource)
+        processLibraryFiles(submission, fileSource)
+        copyFiles(submission.allReferencedFiles(), submission, fileSource)
 
-    private fun generateOutputFiles(submission: ExtendedSubmission) {
-        val simpleSubmission = submission.asSubmission()
-        val json = serializationService.serializeSubmission(simpleSubmission, SubFormat.JSON_PRETTY)
-        val xml = serializationService.serializeSubmission(simpleSubmission, SubFormat.XML)
-        val tsv = serializationService.serializeSubmission(simpleSubmission, SubFormat.TSV)
+        generateOutputFiles(submission.asSubmission(), submission, submission.accNo)
```

Comment:
I see too many concerns in this class and in `processFiles` method we need somehow to make it more specialized (or add at least a TODO for it). One quick idea to come to my mind is to divide in:

1. Class that validate input files, deserialize library files  and set properties in `ExtendedSubmission`
2.  Class that generate output files and store then in disk

Reply:
Done